users/views.py

```diff
@@ -1,3 +1,77 @@
-from django.shortcuts import render
+import json
+import bcrypt
+import jwt
+import re
 
-# Create your views here.
+from django.views         import View
+from django.http          import JsonResponse
+from json.decoder         import JSONDecodeError
+
+from users.models import User
+from my_settings  import SECRET_KEY, ALGORITHM
+
+class SignUpView(View):
+    def post(self, request):
+        try:
+            data         = json.loads(request.body)
+            account      = data["account"]
+            password     = data["password"]
+            email        = data["email"]
+            phone_number = data["phone_number"]
+            nickname     = data.get("nickname")
+            address      = data.get("address")
+
+            email_validation    = re.compile('^[a-z0-9]+@[a-z0-9]+\.[a-z0-9.]+$', re.I)
+            password_validation = re.compile(r'^(?=.*[a-z])(?=.*[0-9])(?=.*[~!@#$%^&*]).{8,}', re.I)
+
+            if not email_validation.match(email):
+                return JsonResponse({"MESSAGE" : "INVALID_EMAIL"}, status=400)
+
+            if not password_validation.match(password):
+                return JsonResponse({"MESSAGE" : "INVALID_PASSWORD"}, status=400)
+           
+            if User.objects.filter(phone_number=data.get('phone_number')).exists()\
+                and data.get('phone_number') != None:
```

Comment:
불필요하게 줄바꿈하지 않아도 괜찮을 것 같네요.
```suggestion
            if User.objects.filter(phone_number=data.get('phone_number')).exists() and ~
```

Reply:
네 수정 반영하겠습니다.

Reply:
줄바꿈하지 않았을 때 통신에러가 뜨는데 혹시 다른 방법을 알려주실 수 있으실까요?